src/systems/systems/src/hydraulic/aerodynamic_model.rs

```diff
@@ -0,0 +1,669 @@
+use nalgebra::{Rotation3, Vector3};
+
+use uom::si::{
+    area::square_meter, f64::*, force::newton, length::meter,
+    mass_density::kilogram_per_cubic_meter,
+};
+
+use crate::{shared::interpolation, simulation::UpdateContext};
+
+/// Projects a 3D box with 3D size along a projection vector.
+/// Outputs the projected area on the plane normal to projection vector
+fn area_projected(size: Vector3<Length>, projection_vector: Vector3<f64>) -> Area {
+    let norm_projection_vector = projection_vector.normalize();
+
+    Area::new::<square_meter>(
+        size[0].get::<meter>() * size[2].get::<meter>() * norm_projection_vector[1]
+            + size[0].get::<meter>() * size[1].get::<meter>() * norm_projection_vector[2]
+            + size[1].get::<meter>() * size[2].get::<meter>() * norm_projection_vector[0],
+    )
+}
```

Comment:
**suggestion**: Always consider what is most important for the person reading the code in this module. Should `area_projected` be the first thing they see, or are other functions or types more important to understand the module? I'd suggest moving this down in the file.

Reply:
As a clever piece of trick from @omryg2 it should deserve to be on top :D  But yeah I can have it lower in the file :p